cmd/snap-seccomp/main_test.go

```diff
@@ -148,21 +153,234 @@ func nativeEndian() binary.ByteOrder {
 	}
 }
 
-// simulateBpf:
+var seccompBpfLoaderContent = []byte(`
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <string.h>
+#include <stdint.h>
+#include <inttypes.h>
+#include <fcntl.h>
+#include <sys/prctl.h>
+
+#include <linux/filter.h>
+#include <linux/seccomp.h>
+
+#define MAX_BPF_SIZE 32 * 1024
+
+int sc_apply_seccomp_bpf(const char *profile_path)
+{
+	unsigned char bpf[MAX_BPF_SIZE + 1]; // account for EOF
+	FILE *fp;
+	fp = fopen(profile_path, "rb");
+	if (fp == NULL) {
+		fprintf(stderr, "cannot read %s\n", profile_path);
+		exit(1);
+	}
+
+	// set 'size' to '1; to get bytes transferred
+	size_t num_read = fread(bpf, 1, sizeof(bpf), fp);
+
+	if (ferror(fp) != 0) {
+		perror("fread()");
+		exit(1);
+	} else if (feof(fp) == 0) {
+		fprintf(stderr, "file too big\n");
+		exit(1);
+	}
+	fclose(fp);
+
+	struct sock_fprog prog = {
+		.len = num_read / sizeof(struct sock_filter),
+		.filter = (struct sock_filter *)bpf,
+	};
+
+	// Set NNP to allow loading seccomp policy into the kernel without
+	// root
+	if (prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0)) {
+		perror("prctl(PR_NO_NEW_PRIVS, 1, 0, 0, 0)");
+		exit(1);
+	}
+
+	if (prctl(PR_SET_SECCOMP, SECCOMP_MODE_FILTER, &prog)) {
+		perror("prctl(PR_SET_SECCOMP, SECCOMP_MODE_FILTER, ...) failed");
+		exit(1);
+	}
+	return 0;
+}
+
+int main(int argc, char *argv[])
+{
+	int rc = 0;
+	if (argc < 2) {
+		fprintf(stderr, "Usage: %s <bpf file> [prog ...]\n", argv[0]);
+		return 1;
+	}
+
+	rc = sc_apply_seccomp_bpf(argv[1]);
+	if (rc || argc == 2)
+		return rc;
+
+	execv(argv[2], (char *const *)&argv[2]);
+	perror("execv failed");
+	return 1;
+}
+`)
+
+var seccompSyscallRunnerContent = []byte(`
+#define _GNU_SOURCE
+#include<unistd.h>
+#include<sys/syscall.h>
+#include<stdlib.h>
+int main(int argc, char **argv) {
+   int l[7];
```

Comment:
The indentation here is different from the one above. We have three spaces here and a tab above, apparently.